src/components/ShareStructure/EditShareStructure.vue

```diff
@@ -280,7 +280,10 @@ export default class EditShareStructure extends Mixins(CurrencyLookupMixin) {
     let rules: Array<Function> = []
     if (!this.hasNoMaximumShares) {
       rules = [
-        (v: string) => /^\d+$/.test(v) || 'Must be a number greater than 0']
+        (v: string) => (v !== null && v !== '') || 'Number of shares is required',
```

Comment:
Can't use `!!v` here because "not not 0" is "true".

Reply:
BTW, v is null before the user enters anything, and '' if the user entered and deleted something.